syntax = "proto3";

service SignUpService {
    rpc signUp (SignUpRequest) returns (SignUpResponse);
}

message SignUpRequest {
    string username = 1;
    string email = 2;
    string password = 3;
}

message SignUpResponse {
    bool boolean = 1;
    string response = 2;
    int32 status = 3;
}


service LoginService {
    rpc login (LoginRequest) returns (LoginResponse);
}

message LoginRequest {
    string email = 1;
    string password = 2;
}

message LoginResponse {
    bool boolean = 1;
    string response = 2;
    int32 status = 3;
}

service TokenService {
    rpc ValidateToken (ValidateTokenRequest) returns (TokenResponse);
}

message ValidateTokenRequest {
    string token = 1;
}

message TokenResponse {
    int32 id = 1;
    string email = 2;
    string password = 3;
    string username = 4;
}